core/controllers/topics_and_skills_dashboard.py

```
@@ -174,6 +177,16 @@ def post(self):
             state_domain.SubtitledHtml.from_dict(explanation_dict))
         skill_services.save_new_skill(self.user_id, skill)
 
+        for filename in filenames:
+            image = self.request.get(filename)
```

what of the request doesn't have filename key, should we raise exception?

---

The images are added alongwith the filenames, so if one is there, the other would be there. Should an exception be thrown here?

---

Yeah I think we should raise 400 instead unexpected 500

---

I think 400, but you could still log an error (logging.error). Check with @kevintab95 what is needed for it to show in StackDriver.

---

It depends on how you're handling the error in the UI. For the error to show up on StackDriver, you can either (a) not handle the failure case in the AJAX call in which case the http interceptor will log the error, or (b) handle the AJAX failure and throw an explicit error with a more descriptive message if required.

---

The `image_validation_service.validate_image_and_filename()` function does take care of this as if this is not declared, 'image' would be None, which is explicitly checked in the function.
So, this would case (a) as @kevintab95 mentioned, right? (not handle failure case in the frontend, but the backend throws the error with message)

---

I think @kevintab95's case (a) refers to throwing an error from the frontend. But here, you want to throw the error in the backend right?

@kevintab95 if we simply do a logging.error() here will it suffice (for showing up in the stackdriver logs)? 

Also, separately -- @aks681 I think you should check for None at the point of get(). If you're passing an "image" and a "filename" to validate_image_and_filename, then I think semantically the image is expected to not be None. Double-checking it there for safety is still a good idea but I think the right place to catch it is actually here (though we can still do both checks, for safety and in case the services function gets used by other methods in the future).

---

Yep, adding a logging.error should suffice in that case.

---

Done, added logging.error.